File: app/src/main/java/app/revanced/bilibili/meta/CouponInfo.java
package app.revanced.bilibili.meta;

import com.alibaba.fastjson.annotation.JSONField;

import java.util.List;

public class CouponInfo {

    @JSONField(name = "list")
    public List<Item> list;

    public CouponInfo(List<Item> list) {
        this.list = list;
    }

    public static class Item {
        @JSONField(name = "type")
        public int type;
        @JSONField(name = "state")
        public int state;
        @JSONField(name = "next_receive_days")
        public long nextReceiveDays;
        @JSONField(name = "vip_type")
        public int vipType;

        public Item(int type, int state, long nextReceiveDays, int vipType) {
            this.type = type;
            this.state = state;
            this.nextReceiveDays = nextReceiveDays;
            this.vipType = vipType;
        }
    }
}

File: integrations/app/src/main/java/app/revanced/bilibili/model/CouponInfo.kt
package app.revanced.bilibili.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class CouponInfo(
    @SerialName("list")
    val list: List<Item> = listOf(),
) {
    @Serializable
    class Item(
        @SerialName("type")
        val type: Int = 0,
        @SerialName("state")
        val state: Int = 0,
        @SerialName("next_receive_days")
        val nextReceiveDays: Long = 0,
        @SerialName("vip_type")
        val vipType: Int = 0
    )
}
